datafusion/expr-common/src/signature.rs

```diff
@@ -343,6 +343,12 @@ impl TypeSignature {
     pub fn supports_zero_argument(&self) -> bool {
         match &self {
             TypeSignature::Exact(vec) => vec.is_empty(),
+            TypeSignature::Any(0) => true,
+            TypeSignature::Variadic(vec) => vec.is_empty(),
```

Comment:
I actually think we could consider removing this function entirely (perhaps deprecate) as it is only used to generate errors